jupyterlab/commands.py

```diff
@@ -979,6 +975,11 @@ def _populate_staging(self, name=None, version=None, public_url=None,
         with open(pkg_path, 'w') as fid:
             json.dump(data, fid, indent=4)
 
+        # copy known-good yarn.lock if missing
+        lock_path = pjoin(staging, 'yarn.lock')
+        if not osp.exists(lock_path):
+            shutil.copy(pjoin(HERE, 'staging', 'yarn.lock'), lock_path)
```

Comment:
> @vidartf  I cannot immediately see the consequences of this

My understanding:

`yarn` in a directory with a checked-in `yarn.lock` and `package.json/(dev)dependencies` will yield the same install every single time... provided a package wasn't pulled from the registry (happens). No solving will occur... this is that happy path that takes less than a second.
- this is where we have been on the shipped application in `/jupyterlab/staging` for over year 

`yarn` in a directory with a `package.json` only will yield a minimum solution `yarn.lock`, with the highest versions available, taking into account all semver ranges. 
- this is where we are today on the user's machine for `lab build` or `labextension install`.
- the delta between what we ship and what users get will grow every day

`yarn` in a directory with a mismatched `yarn.lock` and `package.json`, will bring the `.lock` to consistency with the _minimum possible_ change to existing locked dependencies.
- this is where a user of this PR will be after a `labextension install` before the build

Consider:
- `foo@^1.1.1` was a dependency of core, and is in the shipped lock file as resolving to `1.1.2`, and
- `foo@^1.1.2` is a dependency of `jupyterlab-foo`, which was tested with `foo@1.1.2`
- user installs `jupyterlab-foo`
- user gets just `foo@1.1.2`, hooray :raised_hands:

...but this behavior can create duplicates:
- user installs `jupyterlab-foo`, after `foo@1.1.3` is released
- user gets `foo@1.1.2` (from the lock) and `foo@1.1.3`, boo :ghost:

With `yarn-deduplicate`, the second case will be avoided, and `fewest` and `highest` will both yield `1.1.2`.

However, if `jupyterlab-bar` comes along, and has a dependency chain on `baz -> boo -> buh -> bug -> foo@1.1.2` (which was actually meant to be a devDepenendcy, but it was copy and pasted from `react-container-foo` because of some tutorial from three years ago), then you'll get `1.1.2` and `1.1.3` from `highest`, and `1.1.2` from `fewest`.

Clear as mud, right?

Reply:
I'm guess I'm the one being clear as mud: I meant that I don't see if there are any unintended consequences of this. E.g. will this somehow break the `labextension link` command? I cannot tell, so I wanted to make it clear that I haven't run any such in depth checks.